state-chain/custom-rpc/src/lib.rs

```diff
@@ -2451,8 +2450,25 @@ where
 		dca_parameters: Option<DcaParameters>,
 		at: Option<state_chain_runtime::Hash>,
 	) -> RpcResult<VaultSwapDetails<AddressString>> {
-		self.rpc_backend.with_runtime_api(at, |api, hash| {
-			Ok::<_, CfApiError>(
+		self.rpc_backend.with_versioned_runtime_api(at, |api, hash, api_version| {
+			Ok::<_, CfApiError>(if api_version < 10 {
+				#[expect(deprecated)]
+				api.cf_request_swap_parameter_encoding_before_version_10(
+					hash,
+					broker,
+					source_asset,
+					destination_asset,
+					destination_address.try_parse_to_encoded_address(destination_asset.into())?,
+					broker_commission,
+					try_into_swap_extra_params_encoded(extra_parameters, source_asset.into())?,
+					channel_metadata,
+					boost_fee.unwrap_or_default(),
+					affiliate_fees.unwrap_or_default(),
+					dca_parameters,
+				)??
+				.map_btc_address(Into::into)
+			} else {
+				let network = api.cf_network_environment(hash)?.into();
```

Comment:
It feels a bit weird to have to duplicate all these calls because we improved the interface of the function... But not really a way around it I guess.